http-client/src/main/java/com/tutrit/controller/gateway/CarGatewayHttp.java

```
@@ -0,0 +1,102 @@
+package com.tutrit.controller.gateway;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.tutrit.bean.Car;
+import com.tutrit.gateway.CarGateway;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.Objects;
+import java.util.Optional;
+
+@Component
+public class CarGatewayHttp implements CarGateway {
+    private static final String REQUEST_URI = "http://3.124.0.23:8082/cars/";
+
+    private static final HttpClient httpClient = HttpClient.newBuilder()
+            .version(HttpClient.Version.HTTP_2)
+            .build();
+
+    @Override
+    public Car saveCar(Car car) {
+
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create(REQUEST_URI))
+                .POST(HttpRequest.BodyPublishers.ofString(setBody(car)))
+                .header("Content-Type", "application/json")
+                .build();
+
+        try {
+            httpClient.send(request, HttpResponse.BodyHandlers.ofString());
+        } catch (IOException | InterruptedException e) {
+            Thread.currentThread().interrupt();
```

http connection errors would be ignored in that case, 
you can re-throw RuntimeException

---

Done